guide/english/computer-science/compiled-versus-interpreted-languages/index.md

```diff
@@ -36,13 +36,25 @@ Most programming languages can have both compiled and interpreted implementation
 Strictly speaking, the terms interpreted language and compiled language are not well defined because, in theory, any programming language can be either interpreted or compiled. In modern programming language implementation it is increasingly popular for a platform to provide both options.
 e.g. Python can be executed either as a compiled program, or as an interpreted language in interactive mode.
 
+One of the criteria used to distinguish between a compiled and interpreted language is the presence of a function or statement equivalent to ***eval***, that are absent in stricter, more traditional compiled languages like Java, C and C++. eval is a special function in almost all interpreted languages that accepts valid code in that language as a String, and executes the code as if it had been written in the program normally.
+
+##### Example in JavaScript:
+```JavaScript
+let x = 0;
+eval("x = 'Hello World'");
+console.log(x); //prints Hello World, because x was assigned that value in the eval function
+```
+The use of eval in programs is usually frowned upon, as it can lead to hard to fix errors, and difficult to understand code. From a security standpoint, it can allow the execution of malicious code, so is avoided on websites.
+Modern programs should not need to use the eval function. The behaviour can usually be replicated using a switch statement.
+
 **Most command line tools, CLIs, and shells can theoretically be classified as interpreted languages.**
 
 ### Advantages and Disadvantages
 
 #### Advantages of Compiled Languages
 * Programs compiled into native code at compile time usually tend to be faster than those translated at run time, due to the overhead of the translation process.
 * All the errors are provided to you at once. This allows the programmer to correct all the mistakes at once and recompile the code for execution.
+* Well written programs are usually more robust and less prone to bugs.
```

Comment:
This line doesn't seem like an advantage of compiled languages as such, Well written code is by default less prone to bugs be it in a compiling language or an interpreting language, as a suggestion it would be better to avoid adding this here or maybe add some more description to this to make it a more meaningful advantage.

Reply:
I have added a new section, which I think is the meaning that the line is trying to convey. I am still new to Github, and any advice is appreciated.